File: 1/1.py
from typing import Optional
from common import readList

def get2020Product() -> Optional[int]:
    expense_list = readList('input.txt', int)
    for multiplicand in expense_list:
        for multiplier in expense_list:
            if multiplicand + multiplier == 2020:
                return multiplicand * multiplier
    return None


def main():
    answer = get2020Product()
    print(f'Answer: {answer}')


if __name__ == '__main__':
    main()

File: kotlin/src/main/kotlin/day1/1.kt
package day1

import common.InputReader

class ExpenseReport {
    private val reader = InputReader("day1")
    private val expenseList = reader.getInputListInt()


    fun getProduct(): Int? {
        for (multiplicand in expenseList) {
            for (multiplier in expenseList) {
                if (multiplicand + multiplier == 2020) {
                    return multiplicand * multiplier
                }
            }
        }
        return null
    }


    fun getProductPart2(): Int? {
        for (first in expenseList) {
            for (second in expenseList) {
                for (third in expenseList) {
                    if (first + second + third == 2020) {
                        return first * second * third
                    }
                }
            }
        }
        return null
    }

}


fun main() {
    val report = ExpenseReport()
    val answer1 = report.getProduct()
    val answer2 = report.getProductPart2()

    println("Answer 1: $answer1")
    println("Answer 2: $answer2")
}